checkers-backend/src/service/gamemoves.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Datastore, PropertyFilter } from '@google-cloud/datastore';
+import { gameMove } from 'src/dto/gamemoves.dto'
+import { v4 as uuidv4 } from 'uuid';
+import { ResponseService } from './response.service';
+
+export class GameMovesService {
+    datastore = new Datastore({ databaseId: 'checkers-datastore', projectId: "checkers-zsl" });
+    constructor(private readonly ResponseService: ResponseService) { }
+
+    async findAll(): Promise<gameMove[]> {
+        const query = this.datastore.createQuery("gameMove")
+        const result = await query.run();
+        return result[0];
+    }
+
+    async findOne(id: string): Promise<gameMove> {
+        const query = this.datastore.createQuery("gameMove")
```

Comment:
move query to repository module

Reply:
in this case gamemoves.service.ts should be only a proxy between controller and repository ;)